CONTRIBUTING.md

```diff
@@ -1,221 +1,295 @@
 
-For example let's look at `LineGraphic` in `fastplotlib/graphics/line.py`. Every graphic has a class variable called 
-`_features` which is a set of all graphic properties that are evented. It has the following evented properties: 
-`"data", "colors", "cmap", "thickness"` in addition to properties common to all graphics, such as `"name", "offset", "rotation", and "visible"` 
+We create releases on GitHub, deploy on / distribute via [pypi](https://pypi.org/), and try to follow [semantic versioning](https://semver.org/):
 
-Now look at the constructor for the `LineGraphic` base class `PositionsGraphic`, it first creates an instance of `VertexPositions`. 
-This is a class that manages vertex positions buffer. It defines the line, and provides additional useful functionality. 
-For example, every time that the `data` is changed, the new data will be marked for upload to the GPU before the next draw. 
-In addition, event handlers will be called if any event handlers are registered. 
+> Given a version number MAJOR.MINOR.PATCH, increment the:
+> 1. MAJOR version when you make incompatible API changes
+> 2. MINOR version when you add functionality in a backward compatible manner
+> 3. PATCH version when you make backward compatible bug fixes
 
-`VertexColors`behaves similarly, but it can perform additional parsing that can create the colors buffer from different 
-forms of user input. For example if a user runs: `line_graphic.colors = "blue"`, then `VertexColors.__setitem__()` will 
-create a buffer that corresponds to what `pygfx.Color` thinks is "blue". Users can also take advantage of fancy indexing, 
-ex: `line_graphics.colors[bool_array] = "red"` :smile: 
+To release a new version, we [create a GitHub release](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository) with a new tag incrementing the version as described above. 
+Creating the GitHub release will trigger the deployment to pypi, via our `deploy` action (found in `.github/workflows/pypi-publish.yml`). 
+The built version will grab the version tag from the GitHub release, using [setuptools_scm](https://github.com/pypa/setuptools_scm).
 
-`LineGraphic` also has a `VertexCmap`, this manages the line `VertexColors` instance to parse colormaps, for example: 
-`line_graphic.cmap = "jet"` or even `line_graphic.cmap[50:] = "viridis"`.
+### Testing 
 
-`LineGraphic` also has a `thickness` property which is pretty simple, and `DeletedFeature` which is useful if you need 
-callbacks to indicate that the graphic has been deleted (for example, removing references to a graphic from a legend).
+#### Testing Details
 
-Other graphics have properties that are relevant to them, for example `ImageGraphic` has `cmap`, `vmin`, `vmax`,
-properties unique to images.
+An integral part of our testing suite is not only to test that our code is working, but to also check that are plotting 
+library CI pipeline is producing things that "look visually correct". 
```

Comment:
simplify, something like: "As a plotting library we require two layers of testing. We use a backend test suite that verifies the basic functionality of buffer managers, graphics, layouts, etc., and another test suite which verifies that the library renders plots that are _visually correct_"